From 228a8a073dafe7b057792eafa540e84cf3586b6f Mon Sep 17 00:00:00 2001
From: magic_rb <richard@brezak.sk>
Date: Mon, 2 Jun 2025 01:00:22 +0200
Subject: [PATCH] fix implicit declarations of

---
 src/chkshsgr.c | 1 +
 src/chpst.c    | 1 +
 src/tryshsgr.c | 4 ++++
 3 files changed, 6 insertions(+)

diff --git a/src/chkshsgr.c b/src/chkshsgr.c
index 31eda67..bb0dfce 100644
--- a/src/chkshsgr.c
+++ b/src/chkshsgr.c
@@ -1,5 +1,6 @@
 /* Public domain. */
 
+#include <grp.h>
 #include <sys/types.h>
 #include <unistd.h>
 
diff --git a/src/chpst.c b/src/chpst.c
index d597b68..0140a62 100644
--- a/src/chpst.c
+++ b/src/chpst.c
@@ -3,6 +3,7 @@
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <unistd.h>
+#include <grp.h>
 #include "sgetopt.h"
 #include "error.h"
 #include "strerr.h"
diff --git a/src/tryshsgr.c b/src/tryshsgr.c
index c5ed6d6..08d2fad 100644
--- a/src/tryshsgr.c
+++ b/src/tryshsgr.c
@@ -1,5 +1,9 @@
 /* Public domain. */
 
+#include <sys/types.h>
+#include <unistd.h>
+#include <grp.h>
+
 int main()
 {
   short x[4];
-- 
2.48.1

